conans/client/graph/graph_binaries.py

```diff
@@ -232,29 +195,50 @@ def _process_node(self, node, pref, build_mode, update, remotes):
         if self._evaluate_build(node, build_mode):
             return
 
-        latest_prev_for_pkg_id = self._cache.get_latest_prev(pref)
+        cache_latest_prev = self._cache.get_latest_prev(pref)
+        output = node.conanfile.output
 
-        if latest_prev_for_pkg_id:
-            package_layout = self._cache.pkg_layout(latest_prev_for_pkg_id)
+        if not cache_latest_prev:
+            try:
+                remote, remote_prev, prev_time = self._get_package_from_remotes(node, pref, remotes, update)
+            except NotFoundException:
+                node.binary = BINARY_MISSING
+                node.prev = None
+                node.binary_remote = None
+            else:
+                node.binary = BINARY_DOWNLOAD
+                node.prev = remote_prev.revision
+                node.binary_remote = remote
+        else:
+            package_layout = self._cache.pkg_layout(cache_latest_prev)
             self._evaluate_clean_pkg_folder_dirty(node, package_layout, pref)
-
-        remote = remotes.selected
-        remote_selected = remote is not None
-
-        if not remote:
-            # If the remote_name is not given, follow the binary remote, or the recipe remote
-            # If it is defined it won't iterate (might change in conan2.0)
-            remote_name = self._cache.get_remote(latest_prev_for_pkg_id or pref.ref)
-            remote = remotes.get(remote_name)
-
-        if latest_prev_for_pkg_id:  # Binary already exists in local, check if we want to update
-            node.prev = latest_prev_for_pkg_id.revision
-            self._evaluate_cache_pkg(node, latest_prev_for_pkg_id, remote, remotes, update)
-        else:  # Binary does NOT exist locally
-            # Returned remote might be different than the passed one if iterating remotes
-            remote = self._evaluate_remote_pkg(node, pref, remote, remotes, remote_selected)
-
-        node.binary_remote = remote
+            if update:
+                try:
+                    remote, remote_prev, prev_time = self._get_package_from_remotes(node, pref,
+                                                                                    remotes, update)
+                except NotFoundException:
+                    output.warning("Can't update, no package in remote")
+                except NoRemoteAvailable:
+                    output.warning("Can't update, no remote defined")
```

Comment:
Should this happen? Does it mean that there are no remotes configured or...? <= in that case we should improve the message?